howtos/deployment.rst

```diff
@@ -0,0 +1,348 @@
+
+Under the hood, **AppImage** uses filesystem in user-space 
+(`FUSE <https://github.com/libfuse/libfuse>`_).
+The `packaging process <https://docs.appimage.org/packaging-guide/manual.html#>`__ is pretty straightforward 
+and could be easily automated:
+
+- create a directory like ``MyApp.AppDir``
+- download the `AppImage runtime <https://github.com/AppImage/AppImageKit/releases>`_ (**AppRun** file) and put to the directory 
+- put all dependencies, like libraries (``.so``), resources (e.g. images) inside the directory
+- fill some brief metadata (name, category) to the ``myapp.desktop`` configuration file
+- run **appimagetool** against the directory
+
+As a result, **MyApp-x86_64.AppImage** file will be produced, which is a regular Linux ELF file:
+
+.. code-block:: console
+
+    $ file MyApp-x86_64.AppImage
+    MyApp-x86_64.AppImage: ELF 64-bit LSB executable, x86-64, version 1 (SYSV), dynamically linked, interpreter /lib64/l, for GNU/Linux 2.6.18, stripped
+
+Such a file could be easily distributed just by copying, so it could be uploaded to the FTP server, moved to the flash drive, etc.
+
+Snap
+----
+
+`Snap <https://snapcraft.io/>`_ is the package management system available for the wide range of Linux distributions.
+Unlike ``AppImage``, ``Snap`` requires daemon installed in the system in order to operate. Under the hood, **Snap** is based on `SquashFS <https://github.com/plougher/squashfs-tools>`_.
+
+The `packaging process <https://snapcraft.io/docs/creating-a-snap>`__ is:
+
+- `install <https://snapcraft.io/docs/snapcraft-overview>`_ the snapcraft
+- run ``snapcraft init``
+- edit the ``snap/snapcraft.yml`` `manifest <https://snapcraft.io/docs/snapcraft-format>`_
+- run ``snapcraft`` in order to produce the snap
+- `publish <https://forum.snapcraft.io/t/releasing-your-app/6795>`__ and upload snap, so it could be installed on other systems
+
+Flatpak
+-------
+
+`Flatpak <https://flatpak.org/>`_ is another package management system. Under the hood, **Flatpak** is based on `OSTree <https://ostree.readthedocs.io/en/latest/manual/introduction/>`_.
+
+The `packaging process <http://docs.flatpak.org/en/latest/first-build.html>`__ is:
+
+- install the flatpak runtime and SDK
+- create a manifest ``<app-id>.json``
+- run the ``flatpak-builder`` in order to produce the application
+- `publish <http://docs.flatpak.org/en/latest/publishing.html>`__ the application for further distribution
+
+Alternatively, ``flatpak`` allows distributing the `single-file <http://docs.flatpak.org/en/latest/single-file-bundles.html>`_ package. Such package, however, cannot be run or installed on its own, it's needed to be imported to the local repository on another machine.
+
+Other
+-----
+
+There are enterprise solutions for deployment, which are recommended to be used for production environments, such as 
+`ansible <https://www.ansible.com/>`_, `chef <https://www.chef.io/application-deployment/>`_ and `puppet <https://puppet.com/>`_.
+
+Challenges
```

Comment:
Move this section to "https://docs.conan.io/en/latest/devtools/running_packages.html#running-and-deploying-packages" after the deployment one, you can call it "Deployment challenges"

Reply:
Link somewhere to the customizing settings too with a sentence that transmit the idea, you can define what is your compatibility mode, customizing settings and generating packages for that specific config, they will be different from other packages with a different glibc for instance.

Reply:
And also link back from the customizing settings section to here.

Reply:
I've split pages and added links back and forward to the customizing settings